multiqc/modules/dedup/dedup.py

```diff
@@ -61,16 +62,19 @@ def parse_dedup_log(self, f):
         for k, r in regexes.items():
             r_search = re.search(r, f['f'], re.MULTILINE)
             if r_search:
-                parsed_data[k] = float(r_search.group(1))
-
+                try:
+                    parsed_data[k] = float(r_search.group(1))
+                except ValueError:
+                    parsed_data[k] = r_search.group(1)
         try:
             parsed_data['not_removed'] = parsed_data['total_reads'] - parsed_data['reverse_removed'] - parsed_data['forward_removed'] - parsed_data['merged_removed']
         except KeyError:
             log.debug('Could not calculate "not_removed"')
 
         if len(parsed_data) > 0:
-            # TODO: When tool prints input BAM filename, use that instead
             s_name = self.clean_s_name(os.path.basename(f['root']), f['root'])
+            if 'input_file' in parsed_data:
+                s_name = self.clean_s_name(parsed_data['input_file'], f['root'])
```

Comment:
If you could add an example of this type of file to MultiQC_TestData that would be great!